cmake/west.cmake

```
@@ -1,60 +1,91 @@
-  set(MIN_WEST_VERSION 0.7.1)
-  execute_process(
-    COMMAND
-    ${PYTHON_EXECUTABLE}
-    -c
-    "import west.version; print(west.version.__version__, end='')"
-    OUTPUT_VARIABLE west_version
-    RESULT_VARIABLE west_version_output_result
-    )
 
-  if(WEST_PYTHON)
-    get_filename_component(WEST_PYTHON ${WEST_PYTHON} ABSOLUTE)
-    if(NOT (${WEST_PYTHON} STREQUAL ${PYTHON_EXECUTABLE}))
-      set(PYTHON_EXECUTABLE_OUT_OF_SYNC "\nNote:\n\
-  The Python version used by west is:  ${WEST_PYTHON}\n\
-  The Python version used by CMake is: ${PYTHON_EXECUTABLE}\n\
-  This might be correct, but please verify your installation.")
+if(west_version_output_result)
+  if(DEFINED WEST_PYTHON)
+    if(NOT (${west_realpath} STREQUAL ${python_realpath}))
+      set(PYTHON_EXECUTABLE_OUT_OF_SYNC "\nOr verify these installations:\n\
+  The Python version used by west is:  ${WEST_PYTHON}${west_realpath_msg}\n\
+  The Python version used by CMake is: ${PYTHON_EXECUTABLE}${python_realpath_msg}")
     endif()
-  endif()
 
-  if(west_version_output_result)
-    message(FATAL_ERROR "Unable to import west.version from '${PYTHON_EXECUTABLE}'\n\
-  Please install with:\n\
-      ${PYTHON_EXECUTABLE} -m pip install west\
-  ${PYTHON_EXECUTABLE_OUT_OF_SYNC}")
+    message(FATAL_ERROR "Unable to import west.version from '${PYTHON_EXECUTABLE}':\n${west_version_err}\
+Please install with:\n\
+    ${PYTHON_EXECUTABLE} -m pip install west\
+${PYTHON_EXECUTABLE_OUT_OF_SYNC}")
+  else()
+    # WEST_PYTHON is undefined and we couldn't import west. That's
+    # fine; it's optional.
+    set(WEST WEST-NOTFOUND)
   endif()
+else()
+  # We can import west from PYTHON_EXECUTABLE and have its version.
 
+  # Make sure its version matches the minimum required one.
+  set(MIN_WEST_VERSION 0.7.1)
   if(${west_version} VERSION_LESS ${MIN_WEST_VERSION})
-    message(FATAL_ERROR "The detected west version is unsupported.\n\
-  The version was found to be ${west_version}:\n\
-    ${item}\n\
-  But the minimum supported version is ${MIN_WEST_VERSION}\n\
+    message(FATAL_ERROR "The detected west version, ${west_version}, is unsupported.\n\
+  The minimum supported version is ${MIN_WEST_VERSION}.\n\
   Please upgrade with:\n\
       ${PYTHON_EXECUTABLE} -m pip install --upgrade west\
   ${PYTHON_EXECUTABLE_OUT_OF_SYNC}\n")
   endif()
 
-  # Just output information for a single version. This will still work
-  # even after output is one line.
-  message(STATUS "Found west: ${WEST} (found suitable version \"${west_version}\", minimum required is \"${MIN_WEST_VERSION}\")")
+  # Set WEST to a COMMAND prefix as if it were a find_program() result.
+  if(${west_version} VERSION_GREATER_EQUAL 0.8)
+    # The 'west' package became runnable as a module during 0.8
+    # development.
+    set(WEST_MODULE west)
+  else()
+    # In west 0.7.x, it's possible to run west.app.main. Simplify this
+    # once 0.8.0 is the minimum supported version.
+    set(WEST_MODULE west.app.main)
+  endif()
```

How about only special handling of `0.7.x` like:

```suggestion
set(WEST_MODULE west)
if(${west_version} VERSION_LESS 0.8)
  # In west 0.7.x, west is not directly available as a module so we need to specify full namespace to main.
  string(APPEND WEST_MODULE ".app.main")
endif()
```

Another thing, a clever man once said:
> Same thing here; no need to expose the CLI implementation details. We can just use the west API directly.

(sorry, couldn't resist :laughing: )

---

> How about only special handling of `0.7.x` like:

Sure, that's fine.

Done and done.



> Another thing, a clever man once said:

Heh